circuits/src/zk_gadgets/nonnative.rs

```diff
@@ -1288,14 +1291,51 @@ mod nonnative_tests {
         }
     }
 
+    impl CommitPublic for (BigUint, FieldMod) {
+        type VarType = NonNativeElementVar;
+        type ErrorType = ();
+
+        fn commit_public<CS: RandomizableConstraintSystem>(
+            &self,
+            cs: &mut CS,
+        ) -> Result<Self::VarType, Self::ErrorType> {
+            let expected_words = biguint_to_scalar_words(self.0.clone());
+            let statement_word_vars = expected_words
+                .iter()
+                .map(|word| cs.commit_public(*word))
+                .collect_vec();
+
+            let statement_word_lcs: Vec<LinearCombination> = statement_word_vars
+                .into_iter()
+                .map(Into::into)
+                .collect_vec();
+
+            Ok(NonNativeElementVar::new(statement_word_lcs, self.1.clone()))
+        }
+    }
+
```

Comment:
Similarly to what I did in `lib.rs`, I could make a statement type wrapping a `BigUint` and a `FieldMod` (as it seems that `FieldMod` does belong in the statement, no?) but I opted to just keep it as a `(BigUint, FieldMod)` because of the pending refactor